src/Nethermind/Nethermind.Trie/VisitingOptions.cs

```diff
@@ -39,13 +39,12 @@ public int MaxDegreeOfParallelism
         /// </summary>
         public long FullScanMemoryBudget { get; set; }
 
-        public static int AdjustMaxDegreeOfParallelism(int rawMaxDegreeOfParallelism)
-        {
-            if (rawMaxDegreeOfParallelism == 0)
-                return Math.Max(Environment.ProcessorCount / 4, 1);
-            if (rawMaxDegreeOfParallelism <= -1)
-                return Environment.ProcessorCount;
-            return rawMaxDegreeOfParallelism;
-        }
+        public static int AdjustMaxDegreeOfParallelism(int rawMaxDegreeOfParallelism) =>
+            rawMaxDegreeOfParallelism switch
+            {
+                0 => Math.Max(Environment.ProcessorCount / 4, 1),
```

Comment:
If you surface `Nethermind\Nethermind.Init\Cpu\RuntimeInformation.cs` in some way (perhaps via config) then can use `RuntimeInformation.GetCpuInfo()?.PhysicalCoreCount /2 ?? Environment.ProcessorCount / 4` if that's what you were going for here?